### ISSUE TITLE
HTTPError [AbortError]

### ISSUE THREAD

**Issue by Golorio** [discordjs/discord.js#4311]
**Please describe the problem you are having in as much detail as possible:**
I have been getting the error below. It occurs at random times and seemingly without any spamming from users.
```js
HTTPError [AbortError]: The user aborted a request.
    at RequestHandler.execute (/node_modules/discord.js/src/rest/RequestHandler.js:107:21)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5) {
  code: 500,
  method: 'post',
  path: '/channels/SOMECHANNEL/messages'
}
```

An example when it occurred was when I executed command in a channel that was previously inactive for at least 10 minutes in which the Bot did not respond. After executing a second command the Bot responded and sent both commands at the same time. (I waited on purpose at least 10s between the two commands as to verify it's not a ping issue.) 

**Further details:**

- discord.js version: v12.2.0 (LTS)
- Node.js version: v12.16.3 (LTS)
- Operating system: Centos 7/8
- Priority this issue should have – please be realistic and elaborate if possible: Medium

- [ ] I have also tested the issue on latest master, commit hash:

**Comment by RDambrosio016** [comment]
its a 5xx code, therefore its discord's fault, the api goes down or acts strange occasionally, there isnt much you can do about it.

**Comment by Golorio** [comment]
So isn't it weird that sometimes the Bot won't respond (i.e Discord won't post the message), and on the second command even with 20s delay between them both will get posted at the same time?

**Comment by uhKevinMC** [comment]
@Golorio if it's a reoccurring thing, then that's concerning but if it's happening because of the API, not much we or you can do to solve it.

Reiterating what RDambro said earlier, it is because of the API acting strangely and not responding promptly or at all. It usually lasts no more than an hour, but not much we can do here to solve the issue.

**Comment by TheBlazedBlade** [comment]
Honestly, I've been getting the same issue the last few days (correlating with the time of this post). My bot would load up fine, stay logged in, but would either fail to respond to messages consistently after 10 minutes (the same issue as OP) without shutting down/crashing (no scripts preventing it). So, I'd reload bot via SSH and pm2, as it's hosted on a VPS, to have it once again stop responding within 10 minutes of being idle.

I was questioning if it was my node-modules needing updating, so  I updated to no avail. I wondered if my scripts had errors (as I've been actively working on new code and updating the files), nope no script errors. Next question was my VPS or the VPS framework having issues and started using pm2 to monitor. 

It turns out I've been randomly getting Error 500 errors "HTTPErorr [AbortError]: User Aborted a request" on both GET /gateway/bot and POST /channels/<channel.id>/messages (see attached)

So, looked up the Discord API Status, despite a few fluctuations, seemed within the normal range and no correlation to the 500 errors. In turn, I looked up the HTTPError and turned this issue. It was essentially showing the same problems. 

Info on my bot:

- Ubuntu 16.04.6 LTS (GNU/Linux 4.04.0-178-generic x86_64)

- discord.js@12.2.0 

TL;DR It seems to be a shared, intermittent Error 500 API issue.  

![channelPOSTError](https://user-images.githubusercontent.com/20383282/82900353-be53c980-9f9f-11ea-84d0-2c382c49a392.png)

![errorGatewayBot](https://user-images.githubusercontent.com/20383282/82900360-c14eba00-9f9f-11ea-8d1b-6ec865f7b0d4.png)

**Comment by douglas-sousa** [comment]
I'm also getting this error. It fails almost every time on localhost but I've never had this issue on production

**Comment by iCrawl** [comment]
Duplicate of #3471
